src/H5FDsubfiling/H5FDioc.c

```diff
@@ -224,7 +224,7 @@ H5FD_ioc_init(void)
 
         /* Check if IOC VFD has been loaded dynamically */
         env_var = getenv(HDF5_DRIVER);
-        if (env_var && !strcmp(env_var, H5FD_IOC_NAME)) {
+        if ((env_var && !strcmp(env_var, H5FD_IOC_NAME) && (strlen(env_value) > 0))) {
```

Comment:
Not strictly necessary, but it may be reasonable to have the `strlen` check in between the `env_var` and `!strcmp(env_var, H5FD_IOC_NAME)` checks. If `!strcmp(env_var, H5FD_IOC_NAME)` is true, there's not really a need to check the length of `env_var` at that point. Checking the string length before the `!strcmp(env_var, H5FD_IOC_NAME)` would also save you a string comparison when the environment variable is empty, though that's not really a big deal.